pkg/datasource/glassnode/glassnodeapi/client.go

```diff
@@ -43,11 +43,6 @@ func (c *RestClient) Auth(apiKey string) {
 }
 
 func (c *RestClient) NewRequest(ctx context.Context, method string, refURL string, params url.Values, payload interface{}) (*http.Request, error) {
```

Comment:
requestgen now provides a BaseApiClient, it implements the NewRequest method, so you can reduce your code.

Reply:
check this https://github.com/c9s/requestgen/blob/main/example/api/client.go#L27

Reply:
done